packages/react/src/components/DesignLibrary/DesignLibraryServer.tsx

```diff
@@ -0,0 +1,128 @@
+ * when generation is enabled (`page.mode.designLibrary.isVariantGeneration === true`),
+ * wires the **variant generation** handshake so the parent (DL Studio) can send
+ * generated code to preview and iterate on.
+ * Also Renders the DesignLibraryClientEvents component which serves as a communication bridge between DesignLibraryServer and the Design Studio on the client side.
+ * @param {DesignLibraryServerProps} [props]
+ * @param {Page} [props.page] the page data.
+ * @param {Record<string, DynamicComponent>} [props.componentMap] Component Map will be used by the placeholder to map Sitecore component names to app implementation
+ * @param {ComponentRendering} [props.rendering] Rendering data to be used when rendering the placeholder.
+ * @param {() => Promise<{ default: import('../codegen').ImportEntry[] }>} [props.loadImportMap] Optional async loader that resolves to the import-map used to resolve the generated component’s imports. Required when `isVariantGeneration` is true.
+ * @returns {JSX.Element} The preview surface, or `null` when not in Design Library mode.
+ */
+export const DesignLibraryServer = async ({
+  page,
+  componentMap,
+  rendering,
+  loadImportMap,
+}: DesignLibraryServerProps) => {
+  if (!page.mode.isDesignLibrary) {
+    return null;
+  }
+  let designLibraryStatus = DesignLibraryStatus.READY;
+  let importMap: codegen.ImportEntry[];
+  let importMapInfo: codegen.ImportEntryInfo[];
+  let Component: DynamicComponent;
+  let importMapError: string;
+  let previewComponentStyle: string;
+  const isVariantGeneration = page.mode.designLibrary?.isVariantGeneration;
+
+  // load importmap and importmap payload to pass to FE
+  // if not provided, or errors during load set error to pass to FE
+  if (isVariantGeneration) {
+    if (!loadImportMap) {
+      importMapError = 'No loadImportMap provided';
+    } else {
+      try {
+        const mod = await loadImportMap();
+        importMap = mod.default;
+        importMapInfo = codegen.getImportMapInfo(importMap);
+      } catch (e) {
+        importMapError = `Error loading import map: ${e}`;
+      }
+    }
+  }
+
+  let componentToUpdate = rendering?.placeholders[EDITING_COMPONENT_PLACEHOLDER]?.[0];
+  const componentUpdateKey = `${COMPONENT_UPDATE_CACHE_KEY_PREFIX}${componentToUpdate.uid}`;
+
+  // check if we have an update for this component in the global cache
+  if (hasCache(componentUpdateKey)) {
+    // we have an update, get it and clean the cache
+    designLibraryStatus = DesignLibraryStatus.RENDERED;
+    const updateData = getCacheAndClean<ComponentUpdateModel>(componentUpdateKey);
+
+    // apply the updates to the component rendering
+    if (updateData.updatedComponent) {
+      componentToUpdate.fields = {
+        ...componentToUpdate.fields,
+        ...updateData.updatedComponent.fields,
+      };
+      componentToUpdate.params = {
+        ...componentToUpdate.params,
+        ...updateData.updatedComponent.params,
+      };
```

Comment:
What do you think about introducing a shared function `updateRendering` that will be used here and [there](https://github.com/Sitecore/content-sdk/blob/dev/packages/core/src/editing/design-library.ts#L120-L124). There is a high chance that something will be lost in one of the places in case of some changes or extensions in _updating_ mechanism